src/runtime/components/Modal.vue

```diff
@@ -56,7 +56,9 @@ export interface ModalProps extends DialogRootProps {
   ui?: Partial<typeof modal.slots>
 }
 
-export interface ModalEmits extends DialogRootEmits {}
+export interface ModalEmits extends DialogRootEmits {
+  'after:leave': []
+}
```

Comment:
We need this `after:leave` event so that we can properly unmount the modal after the `leave` animation is finished.